embed.go

```diff
@@ -0,0 +1,61 @@
+package kcp
+
+import (
+	"context"
+	"embed"
+	"path"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsv1client "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	k8serorrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed config
+var configDir embed.FS
+
+// RegisterCRDs registers the CRDs that are in the KCP `config/` directory
+// and are required for the Cluster controller to work correctly.
+// It is useful in all-in-one commands like the kcp, to avoid having to
+// apply them manually
+func RegisterCRDs(cfg *rest.Config) error {
+	crdClient := apiextensionsv1client.NewForConfigOrDie(cfg)
+
+	files, err := configDir.ReadDir("config")
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), "yaml") {
+			continue
+		}
+		bytes, err := configDir.ReadFile(path.Join("config", file.Name()))
+		if err != nil {
+			return err
+		}
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		err = yaml.Unmarshal(bytes, crd)
+		if err != nil {
+			return err
+		}
+
+		_, err = crdClient.CustomResourceDefinitions().Create(context.TODO(), crd, metav1.CreateOptions{})
```

Comment:
style nit:

```
if _, err := crdClient...Create(); k8serrors.IsAlreadyExists(err) {
  ...
} else if err != nil {
  return nil
}
```

This makes it easier to tell at a glance that the error doesn't escape the scope, and won't shadow a previous error.

Same with `err = yaml.Unmarshal` above:

```
if err := yaml.Unmarshal(...); err != nil {
  return err
}
```

This is mainly just hygiene and a good practice to follow, neither change will have a functional effect on this code.

Reply:
Just forced-pushed to fix this.